### PR TITLE
analyzer/fast-sync: Reduce DB contention by writing updates into a writeahead log

### PR DESCRIPTION
**Pull request by mitjat**
Running the fast-sync analyzer with almost any parallelism (as low as parallelism=5) caused significant DB lock contention and consequently slowdowns.
This PR mostly removes the contention by replacing table UPDATEs with INSERTs into new temporary tables in a new `updates_todo` schema. Then after fast-sync, those temp tables get merged back into the "real" tables.

Testing: I locally modified the e2e_regression config to use fast-sync for the majority of the scanned range. Confirmed there were no diffs to the expected output.

Resolves https://app.clickup.com/t/8692uydfe

### CODE REVIEW

File: analyzer/queries/queries.go [hunk 1]
```diff
@@ -92,16 +92,6 @@ var (
     ON CONFLICT (analyzer, height) DO UPDATE SET locked_time = excluded.locked_time
     RETURNING height`
 
-	// IsAnyBlockInRangeProcessedByFastSync = `
```
Comment (mitjat) [inline analyzer/queries/queries.go:95]: Unrelated obsolete code

File: analyzer/consensus/consensus.go [hunk 2]
```diff
@@ -318,11 +339,18 @@ func (m *processor) queueBlockInserts(batch *storage.QueryBatch, data *consensus
 
 func (m *processor) queueEpochInserts(batch *storage.QueryBatch, data *consensusBlockData) error {
 	batch.Queue(
-		queries.ConsensusEpochUpsert,
+		queries.ConsensusFastSyncEpochHeightInsert,
 		data.Epoch,
 		data.BlockHeader.Height,
 	)
```
Comment (Andrew7234) [inline analyzer/consensus/consensus.go:345]: should this only run in fast-sync mode?

File: analyzer/queries/queries.go [hunk 3]
```diff
@@ -554,6 +559,27 @@ var (
...
+	// Recomputes the last round at which a token balance was known to mutate.
+	// Inteded for use after fast-sync.
+	RuntimeEVMTokenBalanceAnalysisMutateRoundRecompute = `
+    INSERT INTO analysis.evm_token_balances AS old
+      (runtime, token_address, account_address, last_mutate_round)
+    (
+      SELECT runtime, token_address, account_address, MAX(last_mutate_round)
+      FROM todo_updates.evm_token_balances
+      WHERE runtime = $1
+      GROUP BY 1, 2, 3
```
Comment (Andrew7234) [inline analyzer/queries/queries.go:577]: nit: 
```suggestion
      GROUP BY runtime, token_address, account_address
```

File: analyzer/queries/queries.go [hunk 4]
```diff
@@ -592,11 +618,27 @@ var (
...
+    INSERT INTO todo_updates.evm_tokens AS old (runtime, token_address, total_supply, num_transfers, last_mutate_round)
+      VALUES ($1, $2, $3, $4, $5)`
+
+	RuntimeEVMTokenRecompute = `
+    INSERT INTO chain.evm_tokens AS old (runtime, token_address, total_supply, num_transfers, last_mutate_round)
+    (
+      SELECT runtime, token_address, SUM(total_supply) AS total_supply, SUM(num_transfers) AS num_transfers, MAX(last_mutate_round) AS last_mutate_round
+      FROM todo_updates.evm_tokens
+      WHERE runtime = $1
+      GROUP BY 1, 2
```
Comment (Andrew7234) [inline analyzer/queries/queries.go:636]:
```suggestion
      GROUP BY runtime, token_address
```

**Review by Andrew7234 — APPROVED** [review]
Looks good overall, thanks!

File: analyzer/consensus/consensus.go (hunk 2 above)
Comment (mitjat) [reply analyzer/consensus/consensus.go]: Yes!! Thank you. I guess technically it wasn't a terrible bug because things would continue to work (as e2e proved also), we'd just create a little extra busy work for the fast-sync finalization stage ... But that's mostly luck :). Fixed!
